modules/guides/guides.routing.yml

```diff
@@ -12,3 +12,4 @@ guides.guide:
     _controller: '\Drupal\guides\Controller\GuidesController::guideContent'
   requirements:
     _permission: 'access guides'
+    _guides_access: 'TRUE'
```

Comment:
I think if we introduce an access checker service then it should also check first whether the current user has the `access guides` permission. So nobody will grant access mistakenly to a path to a user who does not have this permission.